Segfault with latest version + copa 0.10.0

## I get this segfault
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2b7f88a]

goroutine 27 [running]:
github.com/project-copacetic/copacetic/pkg/patch.patchWithContext.func2()
	/home/runner/work/copacetic/copacetic/pkg/patch/patch.go:299 +0x38a
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.11.0/errgroup/errgroup.go:78 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 49
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.11.0/errgroup/errgroup.go:75 +0x93

```

## With this context:

Copa 0.10.0

`podman run -d --rm --name buildkitd --privileged docker.io/moby/buildkit:latest`
`grype docker.io/library/nginx:1.21.6 -o json --file nginx.1.21.6.json`
`copa patch -i docker.io/library/nginx:1.21.6 --scanner grype -r nginx.1.21.6.json -t 1.21.6-patched --addr podman-container://buildkitd`

By the way, grype propose multiple output formats like `cyclonedx-json` what is more standard, so why not base the plugin on it to cover more scanners ?
The doc is also absent to know what is the expected format, I think it's the default `json` but maybe it's the reason of the segfault...

Maybe because your plugin lack some exception handling.

Launching `copa-grype` without argument results on a panic, it seems not good.
```
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
main.main()
	/home/anubhav/Code/copa-grype/copa-grype/copa-grype/main.go:78 +0x11b
```

Also, if the argument is an invalid json we get a plaintext error on stdout instead of stderr
`Error scanning image: invalid character 'x' looking for beginning of value`

Hum... Same error using trivy, what is wrong in my usage ?

```
trivy image docker.io/library/nginx:1.21.6 --format json --scanners vuln --output nginx.1.21.6.json
copa patch -i docker.io/library/nginx:1.21.6 --scanner trivy -r nginx.1.21.6.json -t 1.21.6-patched --addr podman-container://buildkitd

[+] Building 0.0s (0/0)                                                                                                                                                                                                                       
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2b7f88a]

goroutine 25 [running]:
github.com/project-copacetic/copacetic/pkg/patch.patchWithContext.func2()
	/home/runner/work/copacetic/copacetic/pkg/patch/patch.go:299 +0x38a
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.11.0/errgroup/errgroup.go:78 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 33
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.11.0/errgroup/errgroup.go:75 +0x93

```

--- a/README.md
+++ b/README.md
@@ -21,9 +21,14 @@ export PATH=$PATH:/path/to/copagrype/directory
 
 ## Example Usage
 ```shell
+# generate a grype report
+grype <image> -o json --file grype_report.json
+
 # test plugin with example config
 copa-grype grype_report.json
 # this will print the report in JSON format. Example:
 # {"apiVersion":"v1alpha1","metadata":{"os":{"type":"FakeOS","version":"42"},"config":{"arch":"amd64"}},"updates":[{"name":"foo","installedVersion":"1.0.0","fixedVersion":"1.0.1","vulnerabilityID":"VULN001"},{"name":"bar","installedVersion":"2.0.0","fixedVersion":"2.0.1","vulnerabilityID":"VULN002"}]}
 
+# run copa with the scanner plugin (copa-grype) and the report file
+copa patch -i $IMAGE -r grype_report.json --scanner grype
 ```

--- a/main.go
+++ b/main.go
@@ -9,6 +9,11 @@ import (
 )
 
 func main() {
+	if len(os.Args) != 2 {
+        fmt.Fprintf(os.Stderr, "Usage: %s <image_report>\n", os.Args[0])
+        os.Exit(1)
+    }
+
 	// Initialize the parser
 	grypeParser := copaGrype.NewGrypeParser()
 	// Get the image report from command line

--- a/main.go
+++ b/main.go
@@ -21,14 +21,14 @@ func main() {
 
 	report, err := grypeParser.Parse(imageReport)
 	if err != nil {
-		fmt.Printf("Error scanning image: %v\n", err)
+		fmt.Fprintf(os.Stderr, "error scanning image: %v\n", err)
 		return
 	}
 
 	// Serialize the standardized report and print it to stdout
 	reportBytes, err := json.Marshal(report)
 	if err != nil {
-		fmt.Printf("Error serializing report: %v\n", err)
+		fmt.Fprintf(os.Stderr, "error serializing report: %v\n", err)
 		return
 	}
 

